src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -249,6 +249,11 @@ InterpreterSelectQuery::InterpreterSelectQuery(
         source_header = input_pipe->getHeader();
     }
 
+    /// When limits are ignored, also ignore key constraints. It's mainly used to allow mutations to
+    /// work when force_index_by_date or force_primary_key is on.
+    if (options.ignore_limits)
```

Comment:
I don't remember the reason, why we don't just reset these settings when creating Interpreter for mutations?

Reply:
>  why we don't just reset these settings when creating Interpreter for mutations

I don't know, and those are global context settings instead of query context settings.